src/Nest/CommonAbstractions/SerializationBehavior/NestSerializer.cs

```diff
@@ -36,24 +32,28 @@ public NestSerializer(IConnectionSettingsValues settings, JsonConverter stateFul
 
 		public void Serialize(object data, Stream writableStream, SerializationFormatting formatting = SerializationFormatting.Indented)
 		{
+			var defaultFormatting = this._defaultSerializer.Formatting;
+			this._defaultSerializer.Formatting = formatting == SerializationFormatting.Indented ? Formatting.Indented : Formatting.None;
```

Comment:
This is not threadsafe we should probably have a dict of `<SerializationFormatting, Serializer>` that we instantiate in the constructor so that here we can just use it e.g `var serializer  = this._defaultSerializers[formatting]`